src/app/onboarding/basic-information/page.tsx

```diff
@@ -1,95 +1,414 @@
+          <Flex $gap="20px" $w="100%">
+            <FormField
+              control={form.control}
+              name="firstName"
+              render={({ field, fieldState }) => (
+                <FormItem>
+                  <FormLabel>First Name</FormLabel>
+                  <FormControl>
+                    <TextInput
+                      placeholder="Jane"
+                      errorText={fieldState.error?.message}
+                      defaultValue={field.value}
+                      onChange={v => {
+                        onboarding.updateProfile({
+                          first_name: v,
+                        });
+                        field.onChange(v);
+                      }}
+                    />
+                  </FormControl>
+                </FormItem>
+              )}
+            />
+
+            <FormField
+              control={form.control}
+              name="lastName"
+              render={({ field, fieldState }) => (
+                <FormItem>
+                  <FormLabel>Last Name</FormLabel>
+                  <FormControl>
+                    <TextInput
+                      placeholder="Doe"
+                      errorText={fieldState.error?.message}
+                      defaultValue={field.value}
+                      onChange={v => {
+                        onboarding.updateProfile({
+                          last_name: v,
+                        });
+                        field.onChange(v);
+                      }}
+                    />
+                  </FormControl>
+                </FormItem>
+              )}
+            />
+          </Flex>
+
+          <FormField
+            control={form.control}
+            name="country"
+            render={({ field, fieldState }) => (
+              <FormItem>
+                <FormLabel>Country</FormLabel>
+                <FormControl>
+                  <CreatableBigDataDropdown
+                    loadOptions={loadCountries}
+                    error={fieldState.error?.message}
+                    isMulti={false}
+                    onChange={v => {
+                      onboarding.setLocation({
+                        country: v ?? undefined,
+                        state: undefined,
+                        city: undefined,
+                      });
+                      field.onChange(v ?? '');
+                      form.resetField('state', { defaultValue: null });
+                      form.resetField('city', { defaultValue: null });
+                    }}
+                    value={field.value ?? null}
+                    placeholder="Type to search or create a country..."
+                  />
+                </FormControl>
+              </FormItem>
+            )}
+          />
+
+          <FormField
+            control={form.control}
+            name="state"
+            render={({ field, fieldState }) => {
+              const { country } = form.getValues();
```

Comment:
this feels potnetially like a component? not sure though

Reply:
this is more of a render function that the `FormField` takes in; it depends on the actual value of the form, so it there would be little benefit to turn this into a component